rocfile: Use statx to get direct io alignment and size requirements

kurtmcmillan opened the pull request.

kurtmcmillan committed:
rocfile: Wrap statx

kurtmcmillan committed:
rocfile: Use statx (2) to get file status

statx includes direct io alignment requirements and the device IDs for raw devices.

kurtmcmillan committed:
rocfile: Remove stat data from file

Adding statx data to file has made stat data redundant.

kurtmcmillan committed:
rocfile: Use statx direct IO alignment & size data in Fastpath::score()

See statx (2) for more information.

kurtmcmillan committed:
rocfile: IWYU changes

derobins commented:
Are the Jenkins runners missing a package?

```
../../../rocfile/src/backend/fastpath.cpp:132:36: error: use of undeclared identifier 'STATX_DIOALIGN'
```

kurtmcmillan committed:
rocfile/fastpath: Handle STATX_DIOALIGN not being available on pre 6.1 Kernels

If STATX_DIOALIGN is not available fallback to requiring IO size, file offset,
and buffer offset to be multiples of 4KiB.

kurtmcmillan commented:
> Are the Jenkins runners missing a package?
> 
> ```
> ../../../rocfile/src/backend/fastpath.cpp:132:36: error: use of undeclared identifier 'STATX_DIOALIGN'
> ```

No. STATX_DIOALIGN is only available since 6.1. Ubuntu 22, and possibly others, are still running with 5 series. I'll update the PR so that hipFile requires 4kib alignment if STATX_DIOALIGN isn't available.
